lib/kubernetes-test-framework/src/resource_file.rs

```diff
@@ -0,0 +1,27 @@
+use std::path::{Path, PathBuf};
+use tempfile::{tempdir, TempDir};
+
+#[derive(Debug)]
+pub struct ResourceFile {
+    dir: TempDir,
+    path: PathBuf,
+}
+
+impl ResourceFile {
+    pub fn new(data: &str) -> std::io::Result<Self> {
+        let dir = tempdir()?;
+        let path = dir.path().join("custom.yaml");
+        std::fs::write(&path, data)?;
+        Ok(Self { dir, path })
+    }
+
+    pub fn path(&self) -> &Path {
+        self.path.as_path()
+    }
+}
+
+impl Drop for ResourceFile {
+    fn drop(&mut self) {
+        std::fs::remove_file(&self.path).expect("unable to clean up custom resource file");
+    }
+}
```

Comment:
I'm pretty sure this is redundant with the use of a `Tempdir`. It should also delete itself on drop.

Reply:
Good point. I think there was some kind of trouble I had to work around with this, and I didn't really focus on it, and just did it to advance. I'll revisit this part again.

Reply:
Just removing it worked